src/components/PageMetadata.tsx

```diff
@@ -21,230 +21,88 @@ type PropMeta = {
+  description?: string
+  image?: string
+  canonicalUrl?: string
+  author?: string
 }
 
 const PageMetadata: React.FC<IProps> = ({
   description,
-  meta = [],
   title,
   image,
   canonicalUrl,
+  author,
 }) => {
   if (!description) console.warn(`Missing PageMetadata description: ${title}`)
-  const {
-    site,
-    ogImageDefault,
-    ogImageDevelopers,
-    ogImageDapps,
-    ogImageUpgrades,
-  } = useStaticQuery(
-    graphql`
-      {
-        site {
-          siteMetadata {
-            author
-            url
-          }
-        }
-        ogImageDefault: file(relativePath: { eq: "home/hero.png" }) {
-          childImageSharp {
-            gatsbyImageData(
-              width: 1200
-              layout: FIXED
-              placeholder: BLURRED
-              quality: 100
-            )
-          }
-        }
-        ogImageDevelopers: file(relativePath: { eq: "enterprise-eth.png" }) {
-          childImageSharp {
-            gatsbyImageData(
-              width: 1200
-              layout: FIXED
-              placeholder: BLURRED
-              quality: 100
-            )
-          }
-        }
-        ogImageDapps: file(relativePath: { eq: "doge-computer.png" }) {
-          childImageSharp {
-            gatsbyImageData(
-              width: 1200
-              layout: FIXED
-              placeholder: BLURRED
-              quality: 100
-            )
-          }
-        }
-        ogImageUpgrades: file(
-          relativePath: { eq: "upgrades/upgrade_doge.png" }
-        ) {
-          childImageSharp {
-            gatsbyImageData(
-              width: 1200
-              layout: FIXED
-              placeholder: BLURRED
-              quality: 100
-            )
-          }
-        }
-      }
-    `
-  )
 
-  const location = useLocation()
+  const { locale, asPath } = useRouter()
 
-  const { t } = useTranslation()
-  const { language } = useI18next()
+  // TODO: Re-implement intl
+  // const { t } = useTranslation()
 
-  const desc = description || t("site-description")
+  // const desc = description || t("site-description")
+  const desc =
+    description ||
+    "Ethereum is a global, decentralized platform for money and new kinds of applications. On Ethereum, you can write code that controls money, and build applications accessible anywhere in the world."
 
-  const siteTitle = t("site-title")
+  // const siteTitle = t("site-title")
+  const siteTitle = "ethereum.org"
+  const fullTitle = `${title} | ${siteTitle}`
 
   /* Set canonical URL w/ language path to avoid duplicate content */
   /* e.g. set ethereum.org/about/ to ethereum.org/en/about/ */
-  const { pathname } = location
-  let canonicalPath = pathname
-  const firstDirectory = canonicalPath.split("/")[1]
-  if (!isLang(firstDirectory)) {
-    canonicalPath = `/en${pathname}`
-  }
-  const canonical = canonicalUrl || `${site.siteMetadata.url}${canonicalPath}`
+  const canonical = canonicalUrl || join(SITE_URL, "en", asPath)
```

Comment:
Just highlighting the difference here, this would allow `canonicalUrl` to be an empty string... Using `||` would not. Is this the behavior we'd like? The Gatsby repo was using the `||` syntax for these variables.

Reply:
[MDN—Nulish coalescing operator (??)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing):
> The nullish coalescing operator can be seen as a special case of the [logical OR (||) operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_OR). The latter returns the right-hand side operand if the left operand is any [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) value, not only null or undefined. In other words, if you use || to provide some default value to another variable foo, you may encounter unexpected behaviors if you consider some falsy values as usable (e.g., '' or 0).

I think in this case it may be more appropriate to just use `||` to use the fallback in the event of any empty strings. What do you think?

Reply:
@nhsz ^

Reply:
@wackerow agree